ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java

```diff
@@ -138,6 +139,14 @@ public AccountDao(MySqlDataAccessor dataAccessor, MySqlAccountServiceConfig mySq
         String.format("insert into %s (%s, %s, %s, %s, %s, %s, %s, %s) values (?, ?, ?, ?, now(3), ?, ?, ?)",
             DATASET_TABLE, ACCOUNT_ID, CONTAINER_ID, DATASET_NAME, VERSION_SCHEMA, LAST_MODIFIED_TIME, RETENTION_COUNT,
             USER_TAGS, DELETE_TS);
+    //update the dataset field, in order to support partial update, if one parameter is null, keep the original value.
+    //if the delete_ts parameter from input equals to Utils.Infinite_Time(-1), set the delete_ts to null.
+    updateDatasetSql = String.format(
```

Comment:
I have a couple question for this statement.
1. Can we just use IFNULL instead of COALESE(NULLIF(?, ''),XX). If the retentionCount is NULL from the input, we just use the existing retentioinCount, `retentionCount` = IFNULL(?, `retentionCount`)
2. for delete_ts, in the CASE WHEN statement, shouldn't we compare with -1, not '-1', which is a string?

Reply:
We could, I just simply it.
And for delete_ts, since it will compare with timestamp, so it can't be an integer. 
But I do find a bug and fix it after adding more tests. I updated the statement executeUpdateDatasetStatement portion.